.github/workflows/build_iso.yml

```diff
@@ -62,7 +103,8 @@ jobs:
           overwrite: true
 
       - name: Upload ISOs and Checksum to R2
-        if: github.event_name == 'workflow_dispatch'
+        #if: startsWith(github.ref, 'refs/tags/v')
+        if: github.event_name != 'pull_request'
```

Comment:
i think there could be some improvements related to this. As a reusable workflow, you can take inputs, so the input when a callable could set a flag to do this or not. It's a bit awkward at the moment since this file directly run and ALSO reusable, so i'm not sure who we'd get the desireable behavior in a clean way.

This is an example of why i'd suggested making this a reusable workflow that's never run directly. 
Then, when calling from the `build`  workflow, you would pass in `upload_iso: false` but when calling from `build_iso` you could pass `upload_iso: true`.

Also, the logic of determining the "PR specific tag" could go away, because an `override_tag` could be passed in when calling from `build` workflow, but if no override, could just use the `gts` or `latest` tag logic.

For the moment, I do think this works fine, just pointing out where it could be a bit cleaner.

Reply:
Kyle specifically said he never wants to build a container and upload in the same action. Not allowing the workflow to be called directly would not allow Kyle to control when uploads happen.

Reply:
I don't think I explained it well.

My original suggestion was to have a `reusable-build_iso.yml` to which is most of the contents here are moved.

Then `reusable-build_iso.yml` would never run standalone but instead would be called by:
1. `build.yml` pretty much how you have it in this PR
2. `build_iso.yml` a very minimal file similar to what i showed you in ucore, which handles the "normal" ISO builds

The `reusable-build_iso.yml` would define inputs, some examples:
1. `override_tag: string` which could default to empty string or be passed a value
2. `upload_iso: boolean` which must be given a value by the caller

Then `build.yml` would pass `upload_iso: false` and `override_tag: pr-21` (for example), while `build_iso.yml` would pass `upload_iso: true` and no tag override.